### PR TITLE
feat: Add mobile menu

### PR DESCRIPTION
**Pull request by moshoodabdullahi**
# This PR adds the seventh milestone to the portfolio website making use of classList to implement the mobile menu following the layout in [Figma Design](https://www.figma.com/file/LKueaSlr8Kc2yMKeZkKgtm/portfolio_project?type=design&node-id=0-1&t=3pMvwe6OjbChOBN7-0) link of the project and also the [Link](https://github.com/bolah2009/curriculum-html-css/blob/main/portfolio/7_mobile_menu.md) of the mobile menu assignment

### CODE REVIEW

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
plz do give the hamburger icon, the close icon and the links a hover effect aside that youre good to go 😊😊

**Comment by Yinkusadel** [comment]
image
looks fine tho i noticed i can spam your mobile menu till desktop version plz fix that and its all good

**Comment by Yinkusadel** [comment]
image
u seem to have issue with ur links effect here , aside that youre all good

File: Icon - Cancel.svg
Comment (teckcipher) [inline Icon - Cancel.svg:1]: Let's make the file names lower case with not space

File: Icon - Menu.svg
Comment (teckcipher) [inline Icon - Menu.svg:1]: If these icons are not use, we can delete it

File: css/index.css [hunk 1]
```diff
@@ -387,7 +501,64 @@ textarea:focus {
...
+    display: none;
+  }
+
+  .mobile-menu-navbar {
+    color: var(--textarea-border-color);
+    display: flex;
+  }
+
+  .mobile-navbar {
+    font-family: poppins, sans-serif;
```
Comment (teckcipher) [inline css/index.css]: Let's move all font family to CSS variables

File: css/index.css [hunk 2]
```diff
@@ -419,8 +590,8 @@ textarea:focus {
     font: 600 22px poppins, sans-serif;
   }
 
-  .logo {
-    font: 700 24px poppins, sans-serif;
+  .nav-container {
+    font: 700 18px poppins, sans-serif;
```
Comment (teckcipher) [inline css/index.css]: Same here

File: css/index.css [hunk 3]
```diff
@@ -12,23 +12,96 @@
...
+.navbar-menu-li a:hover {
+  cursor: pointer;
+  color: rgb(128 255 0);
+  background-color: #757980;
+}
+
+.navbar-menu-li a:active {
+  cursor: pointer;
+  color: var(--navbar-menu-li-color);
+}
```
Comment (teckcipher) [inline css/index.css]: Let's remove all form of compound selector, using simple selectors make it easier to overwrite styles.

File: css/index.css [hunk 4]
```diff
@@ -37,16 +110,54 @@ body {
...
+  margin-bottom: 8px;
+  font-family: Roboto, sans-serif;
+}
+
+.menu-open-icon {
+  display: flex;
+}
+
+.menu-open-icon:hover {
+  cursor: pointer;
```
Comment (teckcipher) [inline css/index.css]: Why is this element a pointer only when it is hovered? If it's an element that should be pointer then it should with or without hover. I would recommend moving this into general styles. Alternatively, make the element a button and have a general style at the top to make all button elements a pointer cursor and remove all default styles such as background and border.

File: css/index.css [hunk 5]
```diff
@@ -387,7 +501,64 @@ textarea:focus {
...
+}
+
+.mobile-navbar {
+  font-size: 45px;
+}
+
+.d-none {
+  display: none;
+}
+
```
Comment (teckcipher) [inline css/index.css]: Why are we haing two display-none?

File: css/index.css [hunk 6]
```diff
@@ -387,7 +501,64 @@ textarea:focus {
...
+.d-flex {
+  display: flex;
+}
+
+@media (width >=1000px) {
+  .menu-link {
+    display: none;
+  }
```
Comment (teckcipher) [inline css/index.css]: Can we use `d-none-md` here instead? This way we are able to define display none for all medium screens. We can do similar for very large screens by defining `d-none-xl` or `d-none-lg`.

**Review by teckcipher — CHANGES_REQUESTED** [review]
Let's address the comments below.

**Pull request merged**